File: bin/util/yaml2json.py
#!/usr/bin/env python3
#
# yaml2json → Python version
#
# Read a YAML document on stdin and print its equivalent JSON.
#

import json
import sys
import yaml

# https://pyyaml.org/wiki/PyYAMLDocumentation
json.dump(yaml.load(sys.stdin, Loader=yaml.CLoader), sys.stdout, indent=4)
print()

File: bin/util/yaml2json.pl
#!/usr/bin/env perl
#
# yaml2json → perl version
#
# Read several YAML documents and print the equivalent JSON.
#

# https://www.commandlinefu.com/commands/view/12218/convert-yaml-to-json
use YAML::XS qw( LoadFile );
use JSON::XS qw( encode_json );
for (@ARGV) {
	for (LoadFile($_)) {
		print encode_json($_),"\n"
	}
}
